tests/integration/ceph_auth_keystone_test.go

```diff
@@ -0,0 +1,192 @@
+// instead of implementing mocks.
+// The keystone container image used is provided by the Yaook-project
+// (see https://yaook.cloud).
+// The keystone configuration uses sqlite as the database backend.
+// cert-manager and trust-manager are used to provide the certificates
+// (cert-manager) and distribute the ca-certificate file(s) using a secret
+// (trust-manager).
+// To prevent hardcoded credentials, passwords are generated by the tests.
+// The integration tests use the aws s3 client (s3-functionality) and the
+// OpenStack CLI client (keystone- and swift-functionality).
+// (see https://docs.openstack.org/python-openstackclient/latest/).
+// This was a conscious design decision to use the standard client tooling
+// as close as possible to the used instead of using other go-libraries
+// (such as gophercloud) to (re-)implement the client tooling for the test.
+// ***************************************************
+
+func TestCephKeystoneAuthSuite(t *testing.T) {
+	s := new(KeystoneAuthSuite)
+	defer func(s *KeystoneAuthSuite) {
+		HandlePanics(recover(), s.TearDownSuite, s.T)
+	}(s)
+	suite.Run(t, s)
+}
+
+type KeystoneAuthSuite struct {
+	suite.Suite
+	helper    *clients.TestClient
+	installer *installer.CephInstaller
+	settings  *installer.TestCephSettings
+	k8shelper *utils.K8sHelper
+}
+
+func (h *KeystoneAuthSuite) SetupSuite() {
+	namespace := "keystoneauth-ns"
+	h.settings = &installer.TestCephSettings{
+		Namespace:         namespace,
+		OperatorNamespace: namespace,
+		StorageClassName:  "",
+		UseHelm:           true,
+		UsePVC:            false,
+		Mons:              1,
+		SkipOSDCreation:   false,
+		// EnableAdmissionController: true,
+		EnableDiscovery:      true,
+		ChangeHostName:       true,
+		ConnectionsEncrypted: true,
+		RookVersion:          installer.LocalBuildTag,
+		CephVersion:          installer.QuincyVersion,
+		SkipClusterCleanup:   false,
+		SkipCleanupPolicy:    false,
+	}
+	h.settings.ApplyEnvVars()
+	h.installer, h.k8shelper = StartTestCluster(h.T, h.settings)
+
+	// install yaook-keystone here
+	if err := InstallKeystoneInTestCluster(h.k8shelper, namespace); err != nil {
+		return
+	}
+
+	// create usersecret for object store to use
+	testCtx := context.TODO()
+
+	secrets := map[string][]byte{
+		"OS_AUTH_TYPE":            []byte("password"),
+		"OS_IDENTITY_API_VERSION": []byte("3"),
+		"OS_PROJECT_DOMAIN_NAME":  []byte("Default"),
+		"OS_USER_DOMAIN_NAME":     []byte("Default"),
+		"OS_PROJECT_NAME":         []byte(testuserdata["rook-user"]["project"]),
+		"OS_USERNAME":             []byte(testuserdata["rook-user"]["username"]),
+		"OS_PASSWORD":             []byte(testuserdata["rook-user"]["password"]),
+	}
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "usersecret",
+			Namespace: namespace,
+		},
+		Data: secrets,
+	}
+
+	_, err := h.k8shelper.Clientset.CoreV1().Secrets(namespace).Create(testCtx, secret, metav1.CreateOptions{})
+
+	if err != nil {
+		return
+	}
```

Comment:
Instead of returning as though successful, how do we fail this step?

```suggestion
	_, err := h.k8shelper.Clientset.CoreV1().Secrets(namespace).Create(testCtx, secret, metav1.CreateOptions{})
	if err != nil {
		return
	}
```

Reply:
Addressed in 64eafbac648ef8d80784561b51f950bc2f85d9fa